layer.md

```diff
@@ -327,6 +334,38 @@ Implementations SHOULD NOT upload layers tagged with this media type; however, s
 
 [Descriptors](descriptor.md) referencing non-distributable layers MAY include `urls` for downloading these layers directly; however, the presence of the `urls` field SHOULD NOT be used to determine whether or not a layer is non-distributable.
 
+
+# Layer Encryption
+
+To be able to protect the confidentiality of the data in layers, encryption of the layer data blobs can be done to prevent unauthorized access to layer data. Encryption is performed on the data blob of a layer by specifying a mediatype with the `+enc` suffix. For example, `application/vnd.oci.image.layer.v1.tar+enc` is an layer representation of an encrypted `application/vnd.oci.image.layer.v1.tar` layer. 
+
+When using the `+enc` mediatype, the layer data blobs are encrypted with a symmetric encryption algorithm (i.e. AES_128_GCM, AES_128_CBC, etc.) according to the [IANA Registry](https://www.iana.org/assignments/aead-parameters/aead-parameters.xhtml). 
+
+Details of the algorithms and protocols used in the encryption of the data blob are defined in a JSON object below. We note that:
```

Comment:
If I'm understanding this correctly, this structure ends up in the annotations below. It might be helpful to describe the annotations first, then their contents.

I also think for annotations, they need to be added to the annotations.md.

Reply:
Agreed on adding to `annotations.md`. 

 

For the order of explanation, I've found explaining it via the process of the actual encryption works well for users who are not as familiar with encryption.

Here's a stab at phrasing it the other way. 


# Layer Encryption

To be able to protect the confidentiality of the data in layers, encryption of the layer data blobs can be done to prevent unauthorized access to layer data. Encryption is performed on the data blob of a layer by specifying a mediatype with the `+enc` suffix. For example, `application/vnd.oci.image.layer.v1.tar+enc` is an layer representation of an encrypted `application/vnd.oci.image.layer.v1.tar` layer.

When using the `+enc` mediatype, the layer data blobs are encrypted with a symmetric encryption algorithm (i.e. AES_128_GCM, AES_128_CBC, etc.) according to the [IANA Registry](https://www.iana.org/assignments/aead-parameters/aead-parameters.xhtml). These symmetric keys and metadata   of the encryption algorithms used are [wrapped](https://en.wikipedia.org/wiki/Key_Wrap) via an asymmetric key so that only authorized users are able to obtain the symmetric key to decrypt the layer.

The symmetric key and the encryption metadata is defined as the [LayerBlockCipherOptions](specs-go/v1/encryption.go).  The LayerBlockCipherOptions are wrapped in accordance with encrypted message standards such as [OpenPGP(RFC4880)](https://tools.ietf.org/html/rfc4880),                [PKCS7(RFC2315)](https://tools.ietf.org/html/rfc2315), [JWE(RFC7516)](https://tools.ietf.org/html/rfc7516).

The following annotations are used to communicate the encrypted symmetric keys:
- `org.opencontainers.image.enc.keys.pkcs7` - An array of base64 comma separated encrypted messages that contain LayerBlockCipherOptions to perform decryption of the layer data in accordance with [PKCS7(RFC2315)](https://tools.ietf.org/html/rfc2315)
- `org.opencontainers.image.enc.keys.jwe` - An array of base64 comma separated encrypted messages that contain LayerBlockCipherOptions to perform decryption of the layer data in accordance with [JWE(RFC7516)](https://tools.ietf.org/html/rfc7516)
- `org.opencontainers.image.enc.keys.openpgp` - An array of base64 comma separated encrypted messages that contain LayerBlockCipherOptions to perform decryption of the layer data in accordance with [OpenPGP(RFC4880)](https://tools.ietf.org/html/rfc4880)
- `org.opencontainers.image.enc.keys.*` - An array of base64 comma separated encrypted messages that contain LayerBlockCipherOptions to perform decryption of the layer data in accordance with an appropriate standard of specified protocol.

These annotations contains an encrypted message containing the LayerBlockCipherOptions object.

The LackerBlockCipherOPtions object is defined by the JSON object below. We note that:
- The `cipher` field specifies the encryption algorithm in accordance with the [IANA Registry](https://www.iana.org/assignments/aead-parameters/aead-parameters.xhtml).
- The `symkey` field specifies the base64 encoded bytes of the symmetric key used in decryption.
- The `cipherOptions` field specifies additional parameters used in the decryption process of the specified algorithm. This should be in accordance with the RFC standard of the algorithm used.
```
{
    "cipher": "AES_128_GCM",
    "symkey": "54kiln1USEaKnlYhKdz+aA==",
    "cipheroptions": {
        "nonce": "AdcRPTAEhXx6uwuYcOquNA==",
        ...
    }
}
```

Therefore, the decryption of the image can be performed by unwrapping the [LayerBlockCipherOptions](specs-go/v1/encryption.go) structure using the `org.opencontainers.image.enc.keys.*` annotations and using the appropriate cipher with the unwrapped LayerBlockCipherOptions to decrypt   the layer data blob.